src/Uno.UI/UI/Xaml/Documents/TextFormatting/FontDetailsCache.skia.cs

```diff
@@ -1,60 +1,162 @@
 ﻿#nullable enable
 using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
 using HarfBuzzSharp;
 using SkiaSharp;
-using Uno;
+using Uno.Extensions;
 using Uno.Foundation.Logging;
+using Uno.UI;
 using Uno.UI.Xaml;
+using Uno.UI.Xaml.Media;
 using Windows.ApplicationModel;
+using Windows.Storage;
+using Windows.Storage.Helpers;
 using Windows.UI.Text;
-using Uno.UI;
 
 namespace Microsoft.UI.Xaml.Documents.TextFormatting;
 
 internal static class FontDetailsCache
 {
-	// TODO: Investigate best value to use here. SKShaper uses a constant 512 scale, Avalonia uses default font scale. Not 100% sure how much difference it
-	// makes here but it affects subpixel rendering accuracy. Performance does not seem to be affected by changing this value.
-	private const int FontScale = 512;
+	private record struct FontCacheEntry(
+		string? Name,
+		float FontSize,
+		FontWeight Weight,
+		FontStretch Stretch,
+		FontStyle Style);
+
+	private static readonly ConcurrentDictionary<string, SKTypeface?> _typefaceCache = new();
+	private static Dictionary<FontCacheEntry, FontDetails> _fontCache = new();
+	private static object _fontCacheGate = new();
+
+	internal static void OnFontLoaded(string font, SKTypeface? typeface)
+	{
+		_typefaceCache[font] = typeface;
+		lock (_fontCacheGate)
+		{
+			foreach (var key in _fontCache.Keys)
+			{
+				if (key.Name == font)
+				{
+					if (_fontCache.TryGetValue(key, out var details))
+					{
+						if (typeface is null)
+						{
+							// font load failed.
+							details.LoadFailed();
+						}
+						else
+						{
+							details.Update(typeface);
+						}
+					}
+				}
+			}
+		}
+	}
+
+	private static async Task<SKTypeface?> LoadTypefaceFromApplicationUriAsync(Uri uri, FontWeight weight, FontStyle style, FontStretch stretch)
+	{
+		try
+		{
+			// TODO (This comment should be resolved during code review):
+			// Should the user be responsible for adding ".manifest" to the ms-appx path himself?
+			// The benefit of the user doing so is that we will not have to first check if a manifest exists.
```

Comment:
This comment should be deleted. The current implementation is the way we want to go.

Reply:
@Youssef1313 this comment needs to be deleted before we merge then right?